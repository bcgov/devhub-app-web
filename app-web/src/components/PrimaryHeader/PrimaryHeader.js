import React from 'react';
import PropTypes from 'prop-types';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
// login not being implemented at this time
import Login from '../Auth/Login';
import styled from '@emotion/styled';
import { CUSTOM_BREAKPOINTS } from '../../constants/designTokens';

const Header = styled.header`
  background-color: ${({ theme }) => theme.colors.blue};
  border-bottom: 2px solid ${({ theme }) => theme.colors.yellow};
  padding: 0 12px 0 25px;
  color: #fff;
  display: flex;
  box-shadow: 0 1px 2px 2px rgba(0, 0, 0, 0.25);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 50;
  height: 65px;
  @media screen and (min-width: 600px) {
    padding: 0 65px 0 65px;
  }
`;

const StyledHamburger = styled(Hamburger)`
  ${CUSTOM_BREAKPOINTS.navbar} {
    display: none;
  }
`;

const LoginWrapper = styled.div`
  display: none;

  ${CUSTOM_BREAKPOINTS.navbar} {
    display: inline-block;
  }
`;

const Panel = styled.div`
  flex-grow: 1;
  align-items: center;
  justify-content: flex-end;
  display: flex;
`;

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => {
  return (
    <Header>
      <Banner />
      <Panel>
        <LoginWrapper>
          <Login />
        </LoginWrapper>
        {showHamburger ? <StyledHamburger clicked={hamburgerClicked} /> : null}
      </Panel>
    </Header>
  );
};

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
