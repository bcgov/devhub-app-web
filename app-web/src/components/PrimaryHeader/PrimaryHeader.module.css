.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 0 25px;
    color: #fff;
    display: flex;
    height: 65px;
    box-shadow: 0 1px 2px 2px rgba(0, 0, 0, .25);
    position: relative;
}


@media screen and (min-width: 600px) {
    .PrimaryHeader {
        padding: 0 65px 0 65px;
    }
   
}
  