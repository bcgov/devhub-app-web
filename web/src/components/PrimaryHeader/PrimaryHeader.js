import React from 'react';
import PropTypes from 'prop-types';
import classes from './PrimaryHeader.module.css';
import Banner from '../UI/Banner/Banner';
import Hamburger from '../UI/Hamburger/Hamburger';
// login not being implemented at this time
// import Login from '../Auth/Login/Login';

export const PrimaryHeader = ({ showHamburger, hamburgerClicked }) => (
  <header className={classes.PrimaryHeader}>
    <Banner />
    <div className={classes.Other}>
      {/* <Login /> */}
      {showHamburger ? (
        <Hamburger clicked={hamburgerClicked} className={classes.Hamburger} />
      ) : null}
    </div>
  </header>
);

PrimaryHeader.propTypes = {
  showHamburger: PropTypes.bool,
  hamburgerClicked: PropTypes.func,
};

PrimaryHeader.defaultProps = {
  showHamburger: false,
  hamburgerClicked: () => undefined,
};

export default PrimaryHeader;
