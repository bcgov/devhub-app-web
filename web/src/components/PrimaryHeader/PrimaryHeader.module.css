.PrimaryHeader {
    background-color: #036;
    border-bottom: 2px solid #fcba19;
    padding: 0 12px 0 25px;
    color: #fff;
    display: flex;
    box-shadow: 0 1px 2px 2px rgba(0, 0, 0, .25);
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 50;
    height: 65px;
}

.Other {
    flex-grow: 1;
    align-items: center;
    justify-content: flex-end;
    display: flex;
}

.PrimaryHeader .Hamburger {
    cursor: pointer;
}

@media screen and (min-width: 600px) {
    .PrimaryHeader {
        padding: 0 65px 0 65px;
    }
}

@media screen and (min-width: 932px) {
    .PrimaryHeader .Hamburger {
        display: none;
    }
}
