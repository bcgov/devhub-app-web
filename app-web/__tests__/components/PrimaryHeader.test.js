import React from 'react';
import { shallow } from 'enzyme';
import PrimaryHeader from '../../src/components/PrimaryHeader/PrimaryHeader';

describe('Primary Header Component', () => {
  test('it matches snapshot', () => {
    const wrapper = shallow(<PrimaryHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});
