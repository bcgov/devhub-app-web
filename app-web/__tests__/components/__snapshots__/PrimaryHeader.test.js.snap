// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Header Component it matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PrimaryHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Banner
          navigateOnClickPath="/"
          title="Developer's Hub"
        />,
        <Flag
          name="features.login"
        >
          <Connect(Login) />
        </Flag>,
      ],
      "className": "PrimaryHeader",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "navigateOnClickPath": "/",
          "title": "Developer's Hub",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(Login) />,
          "name": "features.login",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Banner
            navigateOnClickPath="/"
            title="Developer's Hub"
          />,
          <Flag
            name="features.login"
          >
            <Connect(Login) />
          </Flag>,
        ],
        "className": "PrimaryHeader",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "navigateOnClickPath": "/",
            "title": "Developer's Hub",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(Login) />,
            "name": "features.login",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
